src/polynomial/univariate.rs

```diff
@@ -39,13 +61,116 @@ impl<F: Field> UnivariatePoly<F> {
                 res + (challenge_in_degree * *coeff)
             })
     }
+
+    pub fn resize(self, new_len: usize) -> Self {
+        if self.len() == new_len {
+            return self;
+        }
+
+        let mut coeff = self.0.into_vec();
+        coeff.resize(new_len, F::ZERO);
```

Comment:
what is the typical use case of this function?  when resize smaller, it might modify the original poly.

Reply:
remastered to `pad_with_zeroes`

Reply:
so the main use case is make it larger